### PR TITLE
188 refactor mappers producer

### PR DESCRIPTION
**Pull request by AshrafAlmohagryDeltares**
Created an ESDLAssetProducerMapper.

### CODE REVIEW

File: src/omotes_simulator_core/adapter/transforms/esdl_asset_mapper.py [hunk 1]
```diff
@@ -60,11 +65,25 @@ def to_entity(model: EsdlAssetObject) -> AssetAbstract:
...
-        return CONVERSION_DICT[type(model.esdl_asset)](
-            model.esdl_asset.name, model.esdl_asset.id, model.get_port_ids()
+
+        # Use the dictionary to get the appropriate mapper
+        asset_type = type(model.esdl_asset)
+        if asset_type in conversion_dict_mappers:
+            mapper = conversion_dict_mappers[asset_type]()
+            return mapper.to_entity(model)
+
+        converted_asset = CONVERSION_DICT[type(model.esdl_asset)](
```
Comment (vanmeerkerk) [inline src/omotes_simulator_core/adapter/transforms/esdl_asset_mapper.py:80]: The two seperate if loops will be removed when all assets have a mapper, right?

File: src/omotes_simulator_core/adapter/transforms/mappers.py [hunk 2]
```diff
@@ -175,8 +175,9 @@ def _convert_assets(self, network: Network) -> List[AssetAbstract]:
             if isinstance(esdl_asset.esdl_asset, esdl_junction):
                 continue
             py_assets_list.append(EsdlAssetMapper.to_entity(esdl_asset))
-            py_assets_list[-1].add_physical_data(esdl_asset=esdl_asset)
             network.add_existing_asset(py_assets_list[-1].solver_asset)
+
+        return py_assets_list
```
Comment (vanmeerkerk) [inline src/omotes_simulator_core/adapter/transforms/mappers.py:180]: CHeck whether there is one return statement; in my review I see two return statements for this method.

**Review by vanmeerkerk — CHANGES_REQUESTED** [review]
I have two small comments, call me if you need clarification.

File: src/omotes_simulator_core/adapter/transforms/esdl_asset_mapper.py (hunk 1 above)
Comment (AshrafAlmohagryDeltares) [reply src/omotes_simulator_core/adapter/transforms/esdl_asset_mapper.py:80]: yes.

**Review by vanmeerkerk — DISMISSED** [review]
Adressed changes

**Review by MichielTukker — APPROVED** [review]

**Pull request merged**